lib/CodeTransform/Tests/Adapter/WorseReflection/Refactor/WorseExpandClassTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Phpactor\CodeTransform\Tests\Adapter\WorseReflection\Refactor;
+
+use Phpactor\CodeTransform\Adapter\WorseReflection\Refactor\WorseExpandClass;
+use Phpactor\CodeTransform\Tests\Adapter\WorseReflection\WorseTestCase;
+use Phpactor\CodeTransform\Domain\SourceCode;
+use Phpactor\Extension\SourceCodeFilesystemExtra\SourceCodeFilestem\Application\ClassSearch;
+use Prophecy\Prophecy\ObjectProphecy;
+
+class WorseExpandClassTest extends WorseTestCase
+{
+    /**
+     * @var ObjectProphecy<ClassSearch>
+     */
+    private ObjectProphecy $classSeach;
+
+    public function setUp(): void
+    {
+        $this->classSeach = $this->prophesize(ClassSearch::class);
+    }
+
+    /**
+     * @dataProvider provideExpandClass
+     */
+    public function testExpandingClass(string $test): void
+    {
+        [$source, $expected, $offset] = $this->sourceExpectedAndOffset(__DIR__ . '/fixtures/' . $test);
+
+        $this->classSeach->classSearch('filesystem', 'Foo')->willReturn([
+            [
+                'file_path' => '/home/mamazu/phpacator/phpactor/SomeNamespace/Foo.php',
+                'class' => 'SomeNamespace\\Foo',
+                'class_name' => 'Foo',
+                'class_namespace' => 'SomeNamespace',
+            ]
+        ])->shouldBeCalled();
```

Comment:
Is that even the correct kind of return value for this class?

Reply:
we should not use `classSearch`, you can do everything with the `Reflector` I think.